feat(react.dev): init

RaiderHCPlay opened the pull request:
<!-- Replace "Website" with a markdown link to the website that you have themed. -->

## 🎉 Theme for React 🎉

![mocha](https://github.com/user-attachments/assets/86a6520b-3fb7-4c6a-b825-ec039326c7f9)
![macchiato](https://github.com/user-attachments/assets/0d8ec984-22c3-4bbc-9510-e87fec4a1395)
![latte](https://github.com/user-attachments/assets/be418f7a-296a-44bf-8776-199ffa695ca2)
![frappe](https://github.com/user-attachments/assets/3a9a9358-1ebc-4615-8363-7190cdc63723)

<!--
You should give a short description of the website that you have themed.
E.g. YouTube is a video sharing platform that allows users to upload, view, and share videos.

You should also attach some screenshots of the themed website, show it off!
-->
React is a popular JavaScript library used for building user interfaces, particularly for web and mobile applications.

## 💬 Additional Comments 💬

<!--
Include any difficulties you had theming this port, or any general comments that would be useful for the reviewer to know.
Feel free to leave this section empty if you don't have anything more to say.
-->

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml) file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.less` - all the CSS for the userstyle, based on the template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

uncenter commented:
None of the generated files should be present. That is all done automatically. Please remove the changes from those scripts.

RaiderHCPlay commented:
I removed the autogenerated files. I hope everything is fine now.

uncenter commented:
You still had lots of changes that can't be made. I squashed all your commits on this branch into one and removed the unwanted changes. In the future, only run commands that the contributing instructions tell you to run. Avoid including/modifying other files, like the `styles/react/catppuccin.user.css` file - that was not needed, and not mentioned in the documentation as something that needed to be added either (it is a "stub" file for a mass file rename we did earlier in January: https://github.com/catppuccin/userstyles/issues/1546).

uncenter commented on scripts/userstyles.yml:
```diff
@@ -905,6 +907,14 @@ userstyles:
     readme:
       app-link: https://app.raindrop.io
     current-maintainers: [*thismoon]
+  react:
+    name: React
```
```suggestion
  react.dev:
    name: React.dev
```

I think this naming will clarify that this port is for React's website/documentation and not for actual use in React code. You will need to rename the `styles/react` directory to `styles/react.dev`, and update the metadata in the comment at the top of the `catppuccin.user.less` file (`deno task lint` should tell you exactly what needs to change).

uncenter commented on scripts/userstyles.yml:
```diff
@@ -905,6 +907,14 @@ userstyles:
+    name: React
+    categories: [development]
+    icon: react
+    color: blue
```
```suggestion
    color: sapphire
```

uncenter commented on scripts/userstyles.yml:
```diff
@@ -905,6 +907,14 @@ userstyles:
+    icon: react
+    color: blue
+    readme:
+      app-link: https://react.dev/
```
```suggestion
      app-link: https://react.dev
```

RaiderHCPlay commented:
I though css file was needed. Sorry about that and thank you.

uncenter commented on styles/react/catppuccin.user.less:
```diff
@@ -0,0 +1,302 @@
+@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
+@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent" ["rosewater:Rosewater", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve*", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+
+==/UserStyle== */
+
+@-moz-document domain("react.dev") {
+
+
+  // change flavor depending on switch toggle
+  html.dark {
+    #catppuccin(@darkFlavor);
+  }
+
+  html {
+    #catppuccin(@lightFlavor);
+  }
+
```
```suggestion
==/UserStyle== */

@-moz-document domain("react.dev") {
  html.dark {
    #catppuccin(@darkFlavor);
  }
  html {
    #catppuccin(@lightFlavor);
  }
```

uncenter commented on styles/react/catppuccin.user.less:
```diff
@@ -0,0 +1,302 @@
+      color: @subtext1;
+    }
+
+    .text-red-50, .text-red-30 {
+      color: @mantle!important;
+    }
```
I have a feeling we shouldn't be changing red text to a shade of gray. Where is this text found originally on the website?

RaiderHCPlay replied:
![image](https://github.com/user-attachments/assets/d923a55d-3ef3-4114-a894-c77676b08897)
It's used here:
https://react.dev/reference/react-dom/components/input#my-checkbox-doesnt-update-when-i-click-on-it

uncenter replied:
We can keep the text color as red and make the background red but with a super low opacity, like they have (e.g. `fade(@red, 10%)`).

RaiderHCPlay replied:
![image](https://github.com/user-attachments/assets/eb716d24-f8dd-4d4e-a719-042e7a408211)
A lot better

uncenter commented on styles/react/catppuccin.user.less:
```diff
@@ -0,0 +1,302 @@
+      color: @subtext0;
+    }
+
+    .text-green-60, .text-green-40, .text-yellow-50, .text-yellow-40, .text-purple-30, .text-purple-50, .text-blue-50, .text-blue-40, .text-blue-30, .text-blue-60 {
+      color: @mantle;
+    }
```
Same here, we should probably actually use `@green`, `@yellow`, etc here instead of reseting them all to mantle.

RaiderHCPlay replied:
![image](https://github.com/user-attachments/assets/8cd42f8b-ef02-41af-9d50-de71953f306d)
![image](https://github.com/user-attachments/assets/9389e1d1-4d5f-4ed7-af19-fba9900e2792)
In my opinion with mantle it looks better.

uncenter commented on styles/react/catppuccin.user.less:
```diff
@@ -0,0 +1,302 @@
+      --sp-colors-surface1: @surface0;
+      --sp-colors-surface2: @surface1;
+      --sp-syntax-color-plain: @text;
+      --sp-syntax-color-comment: @overlay2;
+      --sp-syntax-color-keyword: @sapphire;
+      --sp-syntax-color-tag: @peach;
+      --sp-syntax-color-punctuation: @text;
+      --sp-syntax-color-definition: @accent;
+      --sp-syntax-color-property: @sapphire;
+      --sp-syntax-color-static: @peach;
+      --sp-syntax-color-string: @mauve;
```
```suggestion
      --sp-syntax-color-comment: @overlay2;
      --sp-syntax-color-keyword: @mauve;
      --sp-syntax-color-tag: @blue;
      --sp-syntax-color-punctuation: @overlay2;
      --sp-syntax-color-definition: @yellow;
      --sp-syntax-color-property: @blue;
      --sp-syntax-color-static: @peach;
      --sp-syntax-color-string: @green;
```

uncenter commented:
Please please pull the branch before pushing new changes. You just merged your old branch into the updated branch and now undid all the squashing I had done.

uncenter commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -0,0 +1,296 @@
+    //hover styling
+
+    .hover\:text-link:hover, .text-link:hover, .text-link-dark:hover{
+      color: @sapphire;
```
```suggestion
      color: @accent;
```

uncenter commented:
In the future you can ideally use these buttons to add suggestions, which makes it clearer to understand which suggestions/comments have been resolved.

![image](https://github.com/user-attachments/assets/b1bb704d-acec-477a-b7c8-bcd97a4e1d1c)

RaiderHCPlay commented:
I applied all your requested changes, I fixed every detail I could saw and I updated looking of these color fragments of website:
![image](https://github.com/user-attachments/assets/fb95fbb8-86e8-48be-8f8d-a1ae0b602077)

uncenter commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -0,0 +1,411 @@
+@description Soothing pastel theme for React.dev
+@author Catppuccin
+@license MIT
+
+@preprocessor less
```
```suggestion
@preprocessor less
```

uncenter commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -0,0 +1,411 @@
+
+    // color divs text
+
+    .text-colors(@color) {
+      color: @color !important;
+      svg {
+        color: @color !important;
+      }
+    }
+
+    .text-red-50, .text-red-40 {
+      .text-colors(@red);
+    }
+
+    .text-purple-50, .text-purple-30, .text-purple-60 {
+      .text-colors(@mauve);
+    }
+
+    .text-blue-50, .text-blue-40, .text-blue-60 {
+      .text-colors(@blue);
+    }
+
+    .text-yellow-40, .text-yellow-50, .text-yellow-60 {
+      .text-colors(@yellow);
+    }
```
```suggestion
    #color-and-svg(@color) {
      color: @color !important;
      svg {
        color: @color !important;
      }
    }

    .text-red-50, .text-red-40 {
      #color-and-svg(@red);
    }

    .text-purple-50, .text-purple-30, .text-purple-60 {
      #color-and-svg(@mauve);
    }

    .text-blue-50, .text-blue-40, .text-blue-60 {
      #color-and-svg(@blue);
    }

    .text-yellow-40, .text-yellow-50, .text-yellow-60 {
      #color-and-svg(@yellow);
    }
```

uncenter commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -0,0 +1,411 @@
+      background-color: @base;
+    }
+
+    .color-divs(@divColor) {
```
Let's use a better name here and use an "id" name for the mixin instead of a class (differentiates it more).

uncenter reviewed (changes requested):
Color of the line under code block tabs:

<img width="156" alt="image" src="https://github.com/user-attachments/assets/76d309ea-9d48-493f-9b31-7dffb684a28a" />

Text of social links on https://react.dev/community/team:
![CleanShot 2025-01-31 at 21 14 15](https://github.com/user-attachments/assets/1d259596-d435-491a-b054-6c26a48c46e9)

Down arrow of "deep dive" callouts on https://react.dev/reference/react/useCallback:

![CleanShot 2025-01-31 at 21 14 47](https://github.com/user-attachments/assets/dffa1340-8d13-412d-9494-3220843f567a)
![CleanShot 2025-01-31 at 21 14 53](https://github.com/user-attachments/assets/bf65b943-e7db-4d1a-99c3-6eb6dcac13d8)

Search bar on focus:

![CleanShot 2025-01-31 at 21 15 59](https://github.com/user-attachments/assets/117685c2-e4b7-4524-82d6-d0ef6d691380)

Cancel button in top right while typing in search bar:

![CleanShot 2025-01-31 at 21 16 37](https://github.com/user-attachments/assets/17956ba3-f32a-4d0e-bddb-38ca68319af7)

RaiderHCPlay commented:
I applied requested changes

mxgic1337 commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -0,0 +1,442 @@
+        color: @text;
+      }
+
+      button {
+        background-color: darken(@mauve, 13%) !important;
+
+        svg {
+          color: @text !important;
+        }
+
+      }
```
I'm pretty sure that colors should not be darkened by more than 10%.

```suggestion
      button {
        color: @base !important;
        background-color: @divColor !important;

        svg {
          color: @base !important;
        }
      }
```

uncenter commented:
Last things and then this is good to merge!

Border in these expanded deep dive callouts (again see https://react.dev/reference/react/useCallback):

![CleanShot 2025-02-02 at 12 49 51](https://github.com/user-attachments/assets/bee77ac5-2fca-41f6-b856-e3e9c4a87095)

Below the above pictured deep dives is this tabbed component - the forward/back buttons look unthemed, and the not-current tab border bottom is also unthemed.

![CleanShot 2025-02-02 at 12 50 30](https://github.com/user-attachments/assets/a7ce19a8-2b6e-4742-90ab-c740cbf9826e)

RaiderHCPlay commented:
Changes applied.

uncenter commented:
Two final things!

Links on hover have this underline unthemed in the footer of the root page:

![CleanShot 2025-02-12 at 07 32 18](https://github.com/user-attachments/assets/3f0ac74f-8fea-4a27-b9db-69646c405209)

The "Console" text is unthemed (again in one of the code samples on https://react.dev/reference/react/useCallback):
![CleanShot 2025-02-12 at 07 31 12](https://github.com/user-attachments/assets/799e4270-5f7a-4d29-a0c1-ba01cccdb7e2)

mxgic1337 commented on styles/react.dev/catppuccin.user.less:
```diff
@@ -208,6 +212,12 @@
       border-color: @green;
     }
 
+    .hover\:border-gray-10:hover {
+      &:hover {
+        border-color: @accent;
+      }
+    }
```
```suggestion
    .hover\:border-gray-10:hover {
      border-color: @accent;
    }
```

uncenter commented:
Can you fix the two lint issues? I think I can merge after that :)
